File: transaction-study/src/main/java/yun/blog/transactionstudy/order/OrderApi.java
package yun.blog.transactionstudy.order;

import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/orders")
@RequiredArgsConstructor
public class OrderApi {

  private final OrderService orderService;

  @GetMapping
  public List<Order> get() {
    return orderService.findAll();
  }

  @PostMapping
  public Order create() {
    final Order order = orderService.create();
    return order;

  }



}

File: spring-camp-test/api/src/main/kotlin/com/spring/camp/api/OrderApi.kt
package com.spring.camp.api

import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.math.BigDecimal

@RestController
@RequestMapping("/v1/orders")
class OrderApi {

    @PostMapping
    fun order(
        @RequestBody dto: OrderRequest,
    ): OrderRequest {
        return dto
    }
}

data class OrderRequest(
    val orderNumber: String,
    val status: String,
    val price: Long,
    val address: Address
)

data class Address(
    val zipCode: String,
    val address: String,
    val detail: String,
)